x/examples/fetch/main.go

```diff
@@ -128,19 +210,23 @@ func main() {
 	}
 	resp, err := httpClient.Do(req)
 	if err != nil {
-		log.Fatalf("HTTP request failed: %v\n", err)
+		slog.Error("HTTP request failed", "error", err)
+		os.Exit(1)
 	}
 	defer resp.Body.Close()
 
 	if *verboseFlag {
+		slog.Info("HTTP Proto", "version", resp.Proto)
+		slog.Info("HTTP Status", "status", resp.Status)
 		for k, v := range resp.Header {
-			debugLog.Printf("%v: %v", k, v)
+			slog.Debug("Header", "key", k, "value", v)
 		}
 	}
 
 	_, err = io.Copy(os.Stdout, resp.Body)
 	fmt.Println()
 	if err != nil {
-		log.Fatalf("Read of page body failed: %v\n", err)
+		slog.Error("Read of page body failed", "error", err)
+		os.Exit(1)
```

Comment:
Note that [`os.Exit` will bypass all `defer` calls](https://github.com/golang/go/issues/38261), I guess it's OK in this PR since all resources will be cleaned up when the process ends.

Reply:
Good point. Though I'll also add that this is no different from what we currently have with the Fatal calls, which calls os.Exit.